File: 13-roman-to-integer/13-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        
        int result = 0;
        
        for(int i = 0; i <= s.length() - 1; i++){
            
            if(s.charAt(i) == 'I'){
                try{
                    if(s.charAt(i + 1) == 'V'){
                        result += 4;
                        i += 1;
                    }
                    else if(s.charAt(i + 1) == 'X'){
                        result += 9;
                        i += 1;
                    }
                    else{
                        result += 1;
                    }
                }
                catch (Exception e) {
                    result += 1;
                }
            }
            
            else if(s.charAt(i) == 'X'){
                try{
                    if(s.charAt(i + 1) == 'L'){
                        result += 40;
                        i += 1;
                    }
                    else if(s.charAt(i + 1) == 'C'){
                        result += 90;
                        i += 1;
                    }
                    else{
                        result += 10;
                    }
                }
                catch (Exception e) {
                    result += 10;
                }
            }
            
            else if(s.charAt(i) == 'C'){
                try{
                    if(s.charAt(i + 1) == 'D'){
                        result += 400;
                        i += 1;
                    }
                    else if(s.charAt(i + 1) == 'M'){
                        result += 900;
                        i += 1;
                    }
                    else{
                        result += 100;
                    }
                }
                catch (Exception e) {
                    result += 100;
                }
            }
            
            else if(s.charAt(i) == 'L'){
                    result += 50;
            }
            else if(s.charAt(i) == 'V'){
                    result += 5;
            }
            else if(s.charAt(i) == 'X'){
                    result += 10;
            }
            else if(s.charAt(i) == 'C'){
                    result += 1000;
            }
            else if(s.charAt(i) == 'D'){
                    result += 500;
            }
            else if(s.charAt(i) == 'M'){
                    result += 1000;
            }
        }
        
        return result;
    }
}

File: 13-roman-to-integer/13-roman-to-integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        roman = {
            'I': 1,
            'V': 5,
            'X': 10, 
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        
        result = 0
        
        for i, value in enumerate(s):
            if s[i] == 'V' or s[i] == 'X':
                if s[i - 1] == 'I' and i > 0:
                    result -= 2
            if s[i] == 'L' or s[i] == 'C':
                if s[i - 1] == 'X' and i > 0:
                    result -= 20
            if s[i] == 'D' or s[i] == 'M':
                if s[i - 1] == 'C' and i > 0:
                    result -= 200
            result += roman[value]
        
        return result